fili-core/src/main/java/com/yahoo/bard/webservice/sql/helper/TimeConverter.java

```diff
@@ -0,0 +1,197 @@
+                    put(DefaultTimeGrain.WEEK, asList(YEAR, WEEK));
+                    put(DefaultTimeGrain.DAY, asList(YEAR, DAYOFYEAR));
+                    put(DefaultTimeGrain.HOUR, asList(YEAR, DAYOFYEAR, HOUR));
+                    put(DefaultTimeGrain.MINUTE, asList(YEAR, DAYOFYEAR, HOUR, MINUTE));
+                }
+            };
+
+    /**
+     * Private constructor - all methods static.
+     */
+    private TimeConverter() {
+
+    }
+
+    /**
+     * Gets the number of {@link SqlDatePartFunction} used for the given {@link Granularity}.
+     *
+     * @param granularity  The timegrain to find groupBy functions for.
+     *
+     * @return the number of functions used to groupBy the given granularity.
+     */
+    public static int getNumberOfGroupByFunctions(Granularity granularity) {
+        DefaultTimeGrain timeGrain = getDefaultTimeGrain(granularity);
+        return TIMEGRAIN_TO_GROUPBY.get(timeGrain).size();
+    }
+
+    /**
+     * Builds a list of {@link RexNode} which will effectively groupBy the given {@link Granularity}.
+     *
+     * @param builder  The RelBuilder used with calcite to build queries.
+     * @param granularity  The granularity to build the groupBy for.
+     * @param timeColumn  The name of the timestamp column.
+     *
+     * @return the list of {@link RexNode} needed in the groupBy.
+     */
+    public static Stream<RexNode> buildGroupBy(
+            RelBuilder builder,
+            Granularity granularity,
+            String timeColumn
+    ) {
+        DefaultTimeGrain timeGrain = getDefaultTimeGrain(granularity);
+        return TIMEGRAIN_TO_GROUPBY.get(timeGrain)
+                .stream()
+                .map(sqlDatePartFunction -> builder.call(sqlDatePartFunction, builder.field(timeColumn)));
+    }
+
+    /**
+     * Given a {@link java.sql.ResultSet} and the {@link Granularity} used to make groupBy
+     * statements on time, it will parse out a {@link DateTime} for one row which
+     * represents the beginning of the interval it was grouped on.
+     *
+     *
+     * @param offset the last column before the date fields.
+     * @param row  The results returned by Sql needed to read the time columns.
+     * @param granularity  The granularity which was used when calling
+     * {@link #buildGroupBy(RelBuilder, Granularity, String)}
+     *
+     * @return the datetime for the start of the interval.
+     */
+    public static DateTime parseDateTime(int offset, String[] row, Granularity granularity) {
+        DefaultTimeGrain timeGrain = getDefaultTimeGrain(granularity);
+        DateTime resultTimeStamp = new DateTime(0, DateTimeZone.UTC);
+
+        List<SqlDatePartFunction> times = TIMEGRAIN_TO_GROUPBY.get(timeGrain);
+        for (int i = 0; i < times.size(); i++) {
+            int value = Integer.parseInt(row[offset + i]);
+            SqlDatePartFunction fn = times.get(i);
+            resultTimeStamp = setDateTime(value, fn, resultTimeStamp);
+        }
```

Comment:
I'm not sure I understand the

> That would save us having to use the `SqlDatePartFunction`

Wouldn't we have to use it for the delimiters?

Also is there a delimiter which corresponds to `WEEK`?